dispatcher/config.py

```diff
@@ -0,0 +1,77 @@
+import os
+from contextlib import contextmanager
+from typing import Optional
+
+import yaml
+
+from dispatcher.publish import DispatcherDecorator
+from dispatcher.registry import DispatcherMethodRegistry, registry
+from dispatcher.utils import resolve_callable
+
+
+class DispatcherSettings:
+    def __init__(self, config: dict, task_registry: DispatcherMethodRegistry = registry) -> None:
+        self.brokers: list = config.get('brokers', [])
+        self.producers: list = config.get('producers', [])
+        self.service: dict = config.get('service', {'max_workers': 3})
+        self.callbacks: dict = config.get('callbacks', {})
+        self.tasks: dict = config.get('tasks', {})
```

Comment:
is there a real usecase for tasks defined at this level? I wonder if should try to limit ourselves for the strictly needed funcionality.

Reply:
Writing unit tests against a non-global config would be best practice, and I have some existing tests that use a non-global registry. My thought was that I needed this to test a non-global config _with a non-global registry_.

I agree, it would be best to avoid this. There is an academic reason with the attached `delay` and `apply_async` methods, but more tactically, it would be better to avoid import dependencies. But I need to get the ball rolling on the unit tests, and I won't be confident until then that it'll work out.